### DISCUSSION TITLE
Need help on rpc

### DISCUSSION THREAD

**Discussion by laotala828** [supabase/supabase#8914] [Questions] [answered]
**UPDATE: it turns out that I did not set the RLS correctly. I changed the RLS to allow authenticated users to update and then it worked. Weirdly, I did not get any error message for that.**

I am using RPC to increment a result array. Here is the function

```
create or replace function increment_votes(row_id uuid, newvotes boolean [])
returns void language 'plpgsql' as $$

declare
  i integer;
begin
  FOR i IN SELECT generate_subscripts(newvotes,1)
  LOOP
    if newvotes[i] then
      update polls set results[i]=results[i]+1 where id = row_id;
    end if;
  end LOOP;
end;
$$;
```

When I call it in SQL editor from Supabase dashboard likes this 
```
select increment_votes('887226de-be08-4dcd-bd81-f7beab62dbe5', array[true,false,true,false])
```

it will work, and the results will be updated. Let's say from [0,0,0,0] to [1,0,1,0]. 

But when I use it in my code (+server.ts), it does not update the results. Here is the code.
```
supabase.auth.setAuth(accessToken);

const pollResult = await supabase.rpc('increment_votes', {row_id: pollId, newvotes:[true,true,true,false]});

console.log(pollResult);
```

The console.log will show this
```
{
  error: null,
  data: null,
  count: null,
  status: 204,
  statusText: 'No Content',
  body: null
}
```

I am new to Supabase, am I doing something wrong?

**Comment by steve-chavez** [comment, marked as answer]
Since you have `returns void` in your function it won't return a response body and only a `204 No Content` status, that part should be fine.

> it does not update the results

How are you confirming this?

---

Maybe try using [RAISE](https://www.postgresql.org/docs/current/plpgsql-errors-and-messages.html) to debug the values you pass to the function, then you can query these traces on the `Postgres Logs` UI.

**Reply by laotala828** [reply]
I tried returning some new value from the function, and I did get it from the data and the body properties.

> How are you confirming this?

I checked the value from the table view in the Supabase dashboard. It changed when I ran from SQL editor but did not change when I call the RPC function from the +server.ts file.

Thanks! I will take a look at this one.
> Maybe try using [RAISE](https://www.postgresql.org/docs/current/plpgsql-errors-and-messages.html) to debug the values you pass to the function, then you can query these traces on the `Postgres Logs` UI.

**Reply by steve-chavez** [reply]
Also see https://www.postgresqltutorial.com/postgresql-plpgsql/plpgsql-errors-messages.

Note that `rpc` runs under RLS, check if you have a policy that doesn't allow the UPDATE.

**Reply by laotala828** [reply]
> Also see https://www.postgresqltutorial.com/postgresql-plpgsql/plpgsql-errors-messages.
> 
> Note that `rpc` runs under RLS, check if you have a policy that doesn't allow the UPDATE.

You are right, I changed the RLS to allow authenticated users to update and then it worked. Weirdly, I did not get any error message for that.
